test/util/testfactory/utils.go

```diff
@@ -102,3 +114,211 @@ func GenerateAccounts(count int) []string {
 	}
 	return accs
 }
+
+func NewBaseAccount(kr keyring.Keyring, name string) (*authtypes.BaseAccount, sdk.Coins) {
+	if name == "" {
+		name = tmrand.Str(6)
+	}
+	rec, _, err := kr.NewMnemonic(name, keyring.English, "", "", hd.Secp256k1)
+	if err != nil {
+		panic(err)
+	}
+	addr, err := rec.GetAddress()
+	if err != nil {
+		panic(err)
+	}
+	origCoins := sdk.Coins{sdk.NewInt64Coin(bondDenom, BaseAccountDefaultBalance)}
+	bacc := authtypes.NewBaseAccountWithAddress(addr)
+	return bacc, origCoins
+}
+
+func GenerateBaseAccounts(kr keyring.Keyring, names []string, initExtraCoins ...sdk.Coin) ([]authtypes.BaseAccount, []banktypes.Balance) {
+	bAccounts := make([]authtypes.BaseAccount, len(names))
+	balances := make([]banktypes.Balance, len(names))
+	for i, name := range names {
+		acc, coins := NewBaseAccount(kr, name)
+		bAccounts[i] = *acc
+		balances[i] = banktypes.Balance{
+			Address: acc.GetAddress().String(),
+			Coins:   coins.Add(initExtraCoins...),
+		}
+	}
+
+	return bAccounts, balances
+}
+
+func GenerateDelayedVestingAccounts(kr keyring.Keyring, names []string, initUnlockedCoins ...sdk.Coin) ([]*vestingtypes.DelayedVestingAccount, []banktypes.Balance) {
+	bAccounts, balances := GenerateBaseAccounts(kr, names, initUnlockedCoins...)
+	vAccounts := []*vestingtypes.DelayedVestingAccount{}
+
+	endTime := tmtime.Now().Add(-2 * time.Second)
+	for i := range bAccounts {
+		bal := balances[i]
+		// initUnlockedCoins are subbed to keep them unlocked
+		coins := bal.Coins.Sub(initUnlockedCoins...)
+		acc := vestingtypes.NewDelayedVestingAccount(&bAccounts[i], coins, endTime.Unix())
+		vAccounts = append(vAccounts, acc)
+
+		// the endTime is increased for each account to be able to test various scenarios
+		endTime = endTime.Add(5 * time.Second)
```

Comment:
[blocking]

by adding constant time here that make it almost exclusively useful to the vesting module test. I think in general we need to be focused on creating a generic tooling for creating tooling for installing vesting module accounts and then testing that tooling

Reply:
marked as resolved as the feedback is applied